modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStorageImpl.java

```diff
@@ -1119,7 +1199,16 @@ private void onAckMessage(
         ClusterNode node,
         DistributedMetaStorageUpdateAckMessage msg
     ) {
-        GridFutureAdapter<Boolean> fut = updateFuts.remove(msg.requestId());
+        GridFutureAdapter<Boolean> fut;
+
+        updateFutsStopLock.readLock().lock();
+
+        try {
+            fut = updateFuts.remove(msg.requestId());
+        }
+        finally {
+            updateFutsStopLock.readLock().unlock();
```

Comment:
Readlock looks useless here.
Anyway, you'll finish the future in few lines below.

Reply:
You're right, did that automatically.